rust/fory-derive/src/object/serializer.rs

```diff
@@ -269,72 +271,90 @@ fn generate_default_impl(
 
     match &ast.data {
         Data::Struct(s) => {
-            let fields = sorted_fields(&s.fields);
+            let sorted = sorted_fields(&s.fields);
+            let fields = fields_only(&sorted);
             let is_tuple_struct = super::util::is_tuple_struct(&fields);
 
             use super::util::{
                 classify_trait_object_field, create_wrapper_types_arc, create_wrapper_types_rc,
                 StructField,
             };
 
-            let field_inits: Vec<_> = fields
-                .iter()
-                .map(|field| {
-                    let ident = &field.ident;
-                    let ty = &field.ty;
-
-                    match classify_trait_object_field(ty) {
-                        StructField::RcDyn(trait_name) => {
-                            let types = create_wrapper_types_rc(&trait_name);
-                            let wrapper_ty = types.wrapper_ty;
-                            let trait_ident = types.trait_ident;
-                            let value = quote! {
-                                {
-                                    let wrapper = #wrapper_ty::default();
-                                    std::rc::Rc::<dyn #trait_ident>::from(wrapper)
+            // For tuple structs, we need to order field inits by original index.
+            let field_inits: Vec<_> = if is_tuple_struct {
```

Comment:
If we add a name field to `FieldInfo`(`SortedField` in this pr), could we unify the two branch here?

Introducing two big branch is error-prone, we may forget to update one of rbanch inthe future.